micrometer-core/src/main/java/io/micronaut/configuration/metrics/aggregator/TagsBasedOnMethodInvocationContext.java

```diff
@@ -0,0 +1,11 @@
+package io.micronaut.configuration.metrics.aggregator;
+
+import io.micrometer.core.instrument.Tag;
+import io.micronaut.aop.MethodInvocationContext;
+
+import java.util.List;
+
+public interface TagsBasedOnMethodInvocationContext {
+
+    public List<Tag> buildTags(MethodInvocationContext<Object, Object> context);
```

Comment:
add nullability annotations (`@NonNull` and/or `@Nullable`) to return type and arguments